egs/wsj/s5/steps/segmentation/do_asr_sad_data_dir.sh

```diff
@@ -0,0 +1,261 @@
+#! /bin/bash
+
+# Copyright 2016  Vimal Manohar
+# Apache 2.0.
+
+# This script does nnet3-based speech activity detection given an input kaldi
+# directory and outputs an output kaldi directory.
+# This script can also do music detection and other similar segmentation
+# using appropriate options such as --output-name output-music.
+
+set -e 
+set -o pipefail
+set -u
+
+. path.sh
+
+affix=  # Affix for the segmentation
+nj=32
+cmd=queue.pl
+stage=-1
+
+# Feature options (Must match training)
+mfcc_config=conf/mfcc_hires_bp.conf
+feat_affix=bp   # Affix for the type of feature used
+
+convert_data_dir_to_whole=true    # If true, the input data directory is 
+                                  # first converted to whole data directory (i.e. whole recordings)
+                                  # and segmentation is done on that.
+                                  # If false, then the original segments are 
+                                  # retained and they are split into sub-segments.
+
+# Set to true if the test data needs to be downsampled. 
+# The appropriate sample-frequency is read from the mfcc_config.
+do_resampling=false
+
+output_name=output   # The output node in the network
+sad_name=sad    # Base name for the directory storing the computed loglikes
+                # Can be music for music detection
+segmentation_name=segmentation  # Base name for the directory doing segmentation
+                                # Can be segmentation_music for music detection
+
+# SAD network config
+iter=final  # Model iteration to use
+
+# Contexts must ideally match training for LSTM models, but
+# may not necessarily for stats components
+extra_left_context=0  # Set to some large value, typically 40 for LSTM (must match training)
+extra_right_context=0  
+frames_per_chunk=150
+
+subsampling_factor=1  # Subsampling at the output relative to nnet-output
```

Comment:
Could you explain what this is used for?  Looks like it's multiplied by frame-subsampling-factor.
If you are actually intending to subsample the nnet output, you could just increase the frame-subsampling-factor that you give the network, it will give the same output.  But if this turns out to not be useful right now, I'd prefer not to bake it into the framework- you can always add it later if needed.